Node/core/src/bots/ChatConnector.ts

```diff
@@ -181,95 +191,131 @@ export class ChatConnector implements IConnector, IBotStorage {
             var verifyOptions: jwt.VerifyOptions;
             var openIdMetadata: OpenIdMetadata;
             const algorithms: string[] = ['RS256', 'RS384', 'RS512'];
+            
+            if(this.settings.enableSkills && SkillValidation.isSkillToken(authHeaderValue)) {
+                const skillMsg = utils.clone(req.body);
+                this.prepIncomingMessage(skillMsg);
+
+                let authConfiguration = this.settings.authConfiguration;
+                if(!authConfiguration){
+                    // If the user has not provided an authConfiguration, use the default.
+                    // If the user has not provided allowedCallers then this will throw
+                    authConfiguration = new DefaultAuthenticationConfiguration(this.settings.allowedCallers);
+                }
 
-            if (isEmulator) {
+                JwtTokenValidation.authenticateRequest(
+                    skillMsg, 
+                    authHeaderValue,
+                    new SimpleCredentialProvider(this.settings.appId, this.settings.appPassword),
+                    req.body.serviceUrl,
+                    authConfiguration
+                ).then((claimsIdentity: any) =>{
+                    if (!claimsIdentity || !claimsIdentity.isAuthenticated) {
+                        logger.error('ChatConnector: receive - invalid skill token.');
+                        res.send(403);
+                        res.end();
+                        next();
+                        return; 
+                    }
+                    const oauthScope = JwtTokenValidation.getAppIdFromClaims(claimsIdentity.claims);
+                    const creds = new MicrosoftAppCredentials(this.settings.appId, this.settings.appPassword, oauthScope);
+                    // cache creds in memory
+                    this.credentialsCache[req.body.serviceUrl] = creds
+                    this.dispatch(req.body, res, next);
+                }).catch((err: any) => {
+                    return logger.error(`Error authenticating request: ${err}`)
```

Comment:
do we need to send an error back or do something here? Just logging might mean hanging requests, which means client timeouts :(

Reply:
now logging error, sending 401 status code and call next middleware. i believe this is the appropriate handling.